### PR TITLE
Cleaning up inline documentation

### PR DESCRIPTION
**Pull request by flossypurse**
## What was changed

- updates to doc.go pages
- updates to inline doc comments
- README edits, and adding list of "how-to" guides

## Why?

We want a consistent source of truth for term definitions and developer guides.
Updating the comments to be minimalistic and instead point to the sources of truth in our docs site.

## Checklist

1. Closes 

N/A

2. How was this tested:

N/A

3. Any docs updates needed?

N/A (these are docs changes)

### CODE REVIEW

File: README.md
Comment (cretz) [inline README.md:26]: The `samples library` link shouldn't be a relative path here

File: README.md
Comment (cretz) [inline README.md:41]: Wonder if you want to move this somewhere else. It's strange to just have this one sample in the README and not the dozens of other custom things you could create.

File: worker/worker.go
Comment (cretz) [inline worker/worker.go:25]: This should be removed if you're gonna have a `doc.go`

File: README.md
Comment (cretz) [inline README.md:9]: What are "entities" and "processes" in this context?

File: worker/worker.go
Comment (cretz) [inline worker/worker.go:51]: FWIW, and these docs were written way before me, I think Godoc is clearer to read if it's just capped at 80 chars a line and the sentences come one after another. Adding a newline for each sentence does not affect its output (that whitespace is trimmed).

File: worker/worker.go
Comment (cretz) [inline worker/worker.go:54]: > will wait for a signal from that channel.

Pedantic: "will wait for a receive on that channel."

File: worker/worker.go
Comment (cretz) [inline worker/worker.go:53]: > Pass nil to stop the Worker with an external Stop() call.

Can make people think they couldn't use `Stop()` _unless_ they pass nil. Really you don't have to "Pass nil to stop the Worker with an external Stop() call", you can stop the worker w/ such a call always.

**Comment by flossypurse** [comment]
> @flossypurse - Can you revisit this and update or close if out of date?

Yes, will get back to this soon.

File: README.md
Comment (flossypurse) [reply README.md:41]: Will remove it in this PR, and add it to the docs in this PR: https://github.com/temporalio/documentation/pull/866

**Comment by flossypurse** [comment]
How to use a custom logger is being added to docs here: https://github.com/temporalio/documentation/pull/866

File: README.md
Comment (flossypurse) [inline README.md:41]:
```suggestion
```
duplicate

**Review by cretz — APPROVED** [review]
Looks good to me! Want to get @bergundy's opinion before merging though to confirm what he might like/dislike with this pattern as it might pertain to other SDKs.

File: README.md
Comment (bergundy) [inline README.md:33]: I don't love that we're putting this list here.
Let's keep it in the docs (for now at least).

File: worker/worker.go
Comment (bergundy) [inline worker/worker.go:41]: What does it mean to host tasks?
Why do we need to call it an entity? Feels like we're complicating things.

File: workflow/doc.go
Comment (bergundy) [inline workflow/doc.go:387]: Looks like there's a lot of useful information here that's not found in our docs (e.g. that a side-effect can be executed more than once).

**Review by bergundy — COMMENTED** [review]
I don't love that we're taking useful docs away from the code.
This is the best place for docs because it shows up in IDEs.

Let's talk about it and see if we want to go this direction.

File: README.md
Comment (flossypurse) [reply README.md:26]: > TBH I don't know why we have the list of samples in our docs, seems like it's good enough to put it in the samples repo.

Because internal engineers asked for it. There was also some positive support for this from external devs.

> Are we at least snipsyncing from the samples repo README?

Actually yes, we were. But Snipsync-ing content with clickable links means that docs builds break if the samples repo README links are wrong (makes it harder to update docs urls) - So I recently stopped doing that for the Go samples. We have redirects in docs to handle URL changes instead.

File: README.md
Comment (flossypurse) [reply README.md:33]: So you would prefer a single link?

Comment (bergundy) [reply README.md:33]: SGTM

**Comment by cretz** [comment]
@flossypurse - Are you still wanting to merge this? If so, we can look into solving the conflicts and addressing any comments.

**Comment by flossypurse** [comment]
Thanks for following up @cretz ! No, I think I will close this and circle back with something fresh at another time.

Appreciate it!

**Pull request closed without merging**
